File: leetcode/python/jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, J: str, S: str) -> int:
        count = 0
        d = dict()
        for char in S:
            if char in d:
                d[char] += 1
            else:
                d[char] = 1
        for char in J:
            if char in d:
                count += d[char]
        return count

File: leetcode/java/jewels-and-stones.java
import java.util.HashMap;

class Solution {
    public int numJewelsInStones(String J, String S) {
        int count = 0;
        HashMap<Character, Integer> map = new HashMap<Character, Integer>();

        for (int i = 0; i < S.length(); i++) {
            char c = S.charAt(i);
            int total = map.containsKey(c) ? map.get(c) + 1 : 1;
            map.put(c, total);
        }

        for (int i = 0; i < J.length(); i++) {
            char c = J.charAt(i);
            boolean inMap = map.containsKey(c);
            if (inMap) {
                int keyCount = map.get(c);
                count += keyCount;
            }
        }

        return count;
    }
}